### ISSUE TITLE
Redundent Find pop-up TextBox in the Log History panel doesn't work and needs to be disabled

### ISSUE THREAD

**Issue by slook** [nicotine-plus/nicotine-plus#1501]
Nicotine+ version: 3.1.2.dev1  •  GTK 3.22.11
Operating System/Distribution: Linux MX18, MX19

### Describe the bug
The Log History panel's default GTK Find pop-up TextBox can be activated from anywhere in the main window's with the `Ctrl+F` shortcut key even when the Log History panel is hidden, yet this pop-up TextBox doesn't work so it causes a bad UI obstruction.

A proper Find bar is already well implemented into the Log History, so this little pop-up TextBox is redundant anyway, see screenshot:

![nicotine-plus Redundent Find function doesn't work and should be disabled hidden (Log History panel)](https://user-images.githubusercontent.com/88614182/131335654-3f47fffd-0253-439f-a4cf-f15dce1e0e92.png)

### Expected behavior
More useful keyboard shortcut keys should be implemented to provide access to intended File Search, Filter and Find functions of the main interface that actually work and are useful in context of desired actions, see issue #1500 

### Steps to reproduce the bug

1. Perform a File Search or goto the Downloads tab or Uploads tab
2. Press Ctrl+F on the keyboard
3. Try searching for user content
4. Be frustraited that nothing happened.

It doesn't matter whether the Log History panel is visible or hidden, making this UI bug really obstructive.

### Additional context
To me it looks like some sort of standard feature provided by some GTK widget within the Log panel, that has already been superseded by better custom functionally that does work as expected when the Log panel has the focus, which is fine.

Obviously this annoying pop-up TextBox is just a minor oversight than can hopefully be corrected by disabling the standard GTK feature that makes it appear, see screenshot:

![nicotine-plus Redundent Find function doesn't work and should be disabled (Log History panel)](https://user-images.githubusercontent.com/88614182/131335635-4d6ce627-725d-4dd8-af67-0ed3f850c482.png)

**Comment by mathiascode** [comment]
This is provided by the treeview, and I thought I had disabled it. I'll check.

**Comment by slook** [comment]
Yes, it is fixed in the Search Files tab now :-)

> 1. Perform a File Search or goto the Downloads tab or Uploads tab

Unfortunately @mathiascode , as I said in my OP the same bug also appears in the Downloads tab and Uploads tab. Also in User Browse aswell :-(

**Comment by mathiascode** [comment]
Sorry, been a long day. I wouldn't consider this a bug in the downloads/uploads/user browse tabs, it's just the default search column that's perhaps a bit unfitting. I believe it searches the user column in the downloads/uploads tabs, not sure about user browse.

**Comment by slook** [comment]
No worries @mathiascode . Ah yes you are right I was trying Filenames in the dl / ul tabs that led me to think it didn't work. Indeed it does work for searching Filenames in User Browse, actually it's quite handy there.

BTW it is not my expectation that you will read, reply nor action my queries in any particular timeframe.

**Comment by mathiascode** [comment]
> BTW it is not my expectation that you will read, reply nor action my queries in any particular timeframe.

It's fine, I just prefer to resolve simple matters as soon as possible in order to have a shorter issue list. Too many bad memories from projects with massive backlogs. :)

I've set the filename column as the search column in transfer lists.

**Comment by slook** [comment]
Cool, that's alright then as long as it doesn't bother you @mathiascode . I saw that git got up to 40 open issues and thought twice about posting some of the minor tweaks and stuff.

> I've set the filename column as the search column in transfer lists.

That's a much better idea that makes far more sense, now that Download history is remembered perhaps it will be needed.

It seems like most of the major bugs have gone now, is that why it's quiet on git? Where is everyone else?

Do you or anyone else have any sort of road map for the project? What are your goals?

**Comment by mathiascode** [comment]
> It seems like most of the major bugs have gone now, is that why it's quiet on git? Where is everyone else?

Some people are in the IRC channel or Soulseek private messages, but most of the major stuff is done.

> Do you or anyone else have any sort of road map for the project? What are your goals? Edit: I mean, what is your opinion on the sort of stuff you want to prioritize?

I've accomplished most of my personal needs, and refactored/rewritten almost everything at this point, so it's mostly minor polish and fixes at this point (whatever users happen to report). I don't have any concrete roadmap anymore, but I prioritize bug fixes. My main goal has always been to create a better Soulseek client. :)

**Comment by slook** [comment]
> My main goal has always been to create a better Soulseek client. :)

You have already surpassed that goal for sure!... I believe it's becoming it's own animal now.

> it's mostly minor polish and fixes at this point (whatever users happen to report)

That is very honorable, but what specific components do you personally feel needs working on the most?

**Comment by mathiascode** [comment]
> That is very honorable, but what specific components do you personally feel needs working on the most? Or, what sorts of things to you most enjoy developing, ie: core, UI?

I've disliked our tabs for a long time, but you've inspired me to invest some time in actually improving them. Previously I would've said chats are the most lacking, but they were improved a lot during the last cycle. Chat-related code probably needs the most cleanup still. 

In general, accessibility is the main part that needs work. After that, probably https://github.com/nicotine-plus/nicotine-plus/issues/994 to finally have a full Soulseek protocol implementation.

I like working on both the core and UI tbh (both need to be solid for a good experience), but I like to work in cycles. I'll mostly work on the GUI for some days/weeks, then the core for some time etc.

**Comment by slook** [comment]
That's interesting I must admit that tabs was one of the first things that struck me as being rather clunky ever since version 1, perhaps it gave a bad first impression that's why I overlooked the client for so long in the early days. So I'm glad you are on the same page there, so to speak, and that you've now sorted a keyboard shortcuts solution is ace. Indeed the chat isn't the best is it, although it is very functional there are a couple of minor issues I've noticed like Username copying, URL copying and the whole autocomplete thing is just a pickle it seems!

You might want to edit your previous comment (above) because this issue is irrelevant for 'Implement search request delivery to child peers' in which this thread is now mentioned.

That's cool, a very good stratagy I feel. I must say have really enjoyed ironing out the UI issues with you over the last few days I think it's been a good cycle where a lot of significant progress has been achieved already. I think I start to see your direction with it, and you definitely seem to be going the right way about it. As to the subject of core I know little about but I do find the slsk network to be a highly interesting beast, considering how rudimentary it's underlying design is I find it amazing how powerful it feels with a decent client.
